PWGHF/vertexingHF/AliAnalysisTaskSEXicPlusToXi2PifromKFP.cxx

```diff
@@ -2398,10 +2436,18 @@ void AliAnalysisTaskSEXicPlusToXi2PifromKFP::FillTreeRecXicPlusFromCasc(AliAODEv
   fVar_XicPlus_QA[74] = fVar_XicPlus_QA[72] - fVar_XicPlus_QA[46]; // SV: pt diff of Xic+ (w/ topo. constraint recalPV - PV)
 
   fVar_XicPlus_QA[75] = fpVtx->CountRealContributors(); // PV: count daughter primary tracks
+  fVar_XicPlus_QA[76] = GetMCEventID(); // Event ID for MC
+  }
+  if (!fIsMC) {
+    fVar_XicPlus_QA[76] = GetEventIdAsLong(AODEvent->GetHeader()); // Event ID for Data
```

Comment:
`fVar_XicPlus_QA` is a `Float_t*` so you are rather casting the `ULong64_t` (which has 64 bits) to a `Float_t` (that has 32 bits), so you lose some information and likely your EventID will not be unique